### ISSUE TITLE
Build failing for v1.0.0

### ISSUE THREAD

**Issue by pramodhrachuri** [open-infrastructure-labs/caerus#8]
Hi,
We are trying to rebuild v1.0.0 of caerus-dike and facing problems. When we did a git checkout to `tags/v1.0.0` and tried building it, we got `java.lang.StackOverflowError`. I have added the tail of the log at the end.

Please help us debug this issue. Thanks in advance!
-Pramodh

```           
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spark Project Parent POM 3.3.0-SNAPSHOT:
[INFO] 
[INFO] Spark Project Parent POM ........................... SUCCESS [01:22 min]
[INFO] Spark Project Tags ................................. SUCCESS [ 26.482 s]
[INFO] Spark Project Sketch ............................... SUCCESS [ 10.063 s]
[INFO] Spark Project Local DB ............................. SUCCESS [  8.611 s]
[INFO] Spark Project Networking ........................... SUCCESS [ 13.225 s]
[INFO] Spark Project Shuffle Streaming Service ............ SUCCESS [  8.326 s]
[INFO] Spark Project Unsafe ............................... SUCCESS [ 15.882 s]
[INFO] Spark Project Launcher ............................. SUCCESS [  9.190 s]
[INFO] Spark Project Core ................................. SUCCESS [04:04 min]
[INFO] Spark Project ML Local Library ..................... SUCCESS [ 40.565 s]
[INFO] Spark Project GraphX ............................... SUCCESS [ 43.569 s]
[INFO] Spark Project Streaming ............................ SUCCESS [01:21 min]
[INFO] Spark Project Catalyst ............................. FAILURE [01:05 min]
[INFO] Spark Project SQL .................................. SKIPPED
[INFO] Spark Project ML Library ........................... SKIPPED
[INFO] Spark Project Tools ................................ SKIPPED
[INFO] Spark Project Hive ................................. SKIPPED
[INFO] Spark Project REPL ................................. SKIPPED
[INFO] Spark Project Assembly ............................. SKIPPED
[INFO] Kafka 0.10+ Token Provider for Streaming ........... SKIPPED
[INFO] Spark Integration for Kafka 0.10 ................... SKIPPED
[INFO] Kafka 0.10+ Source for Structured Streaming ........ SKIPPED
[INFO] Spark Project Examples ............................. SKIPPED
[INFO] Spark Integration for Kafka 0.10 Assembly .......... SKIPPED
[INFO] Spark Avro ......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10:52 min
[INFO] Finished at: 2021-11-07T23:18:32Z
[INFO] ------------------------------------------------------------------------
---------------------------------------------------
constituent[0]: file:/usr/share/maven/conf/logging/
constituent[1]: file:/usr/share/maven/lib/sisu-plexus.jar
constituent[2]: file:/usr/share/maven/lib/maven-plugin-api-3.x.jar
constituent[3]: file:/usr/share/maven/lib/jcl-over-slf4j.jar
constituent[4]: file:/usr/share/maven/lib/maven-resolver-connector-basic.jar
constituent[5]: file:/usr/share/maven/lib/maven-resolver-transport-wagon.jar
constituent[6]: file:/usr/share/maven/lib/maven-compat-3.x.jar
constituent[7]: file:/usr/share/maven/lib/maven-slf4j-provider-3.x.jar
constituent[8]: file:/usr/share/maven/lib/maven-resolver-api.jar
constituent[9]: file:/usr/share/maven/lib/guice.jar
constituent[10]: file:/usr/share/maven/lib/plexus-component-annotations.jar
constituent[11]: file:/usr/share/maven/lib/commons-lang3.jar
constituent[12]: file:/usr/share/maven/lib/plexus-utils.jar
constituent[13]: file:/usr/share/maven/lib/maven-core-3.x.jar
constituent[14]: file:/usr/share/maven/lib/maven-shared-utils.jar
constituent[15]: file:/usr/share/maven/lib/jsr250-api.jar
constituent[16]: file:/usr/share/maven/lib/maven-model-builder-3.x.jar
constituent[17]: file:/usr/share/maven/lib/sisu-inject.jar
constituent[18]: file:/usr/share/maven/lib/plexus-cipher.jar
constituent[19]: file:/usr/share/maven/lib/maven-embedder-3.x.jar
constituent[20]: file:/usr/share/maven/lib/cdi-api.jar
constituent[21]: file:/usr/share/maven/lib/plexus-interpolation.jar
constituent[22]: file:/usr/share/maven/lib/wagon-file.jar
constituent[23]: file:/usr/share/maven/lib/slf4j-api.jar
constituent[24]: file:/usr/share/maven/lib/maven-settings-builder-3.x.jar
constituent[25]: file:/usr/share/maven/lib/maven-model-3.x.jar
constituent[26]: file:/usr/share/maven/lib/maven-resolver-util.jar
constituent[27]: file:/usr/share/maven/lib/commons-cli.jar
constituent[28]: file:/usr/share/maven/lib/maven-builder-support-3.x.jar
constituent[29]: file:/usr/share/maven/lib/maven-resolver-impl.jar
constituent[30]: file:/usr/share/maven/lib/wagon-provider-api.jar
constituent[31]: file:/usr/share/maven/lib/jansi.jar
constituent[32]: file:/usr/share/maven/lib/commons-io.jar
constituent[33]: file:/usr/share/maven/lib/maven-settings-3.x.jar
constituent[34]: file:/usr/share/maven/lib/maven-resolver-provider-3.x.jar
constituent[35]: file:/usr/share/maven/lib/javax.inject.jar
constituent[36]: file:/usr/share/maven/lib/guava.jar
constituent[37]: file:/usr/share/maven/lib/wagon-http-shaded.jar
constituent[38]: file:/usr/share/maven/lib/maven-artifact-3.x.jar
constituent[39]: file:/usr/share/maven/lib/maven-resolver-spi.jar
constituent[40]: file:/usr/share/maven/lib/maven-repository-metadata-3.x.jar
constituent[41]: file:/usr/share/maven/lib/aopalliance.jar
constituent[42]: file:/usr/share/maven/lib/plexus-sec-dispatcher.jar
---------------------------------------------------
Exception in thread "main" java.lang.StackOverflowError
	at scala.reflect.internal.Trees.itransform(Trees.scala:1402)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
	at scala.reflect.internal.Trees.itransform(Trees.scala:1409)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
	at scala.reflect.internal.Trees.itransform(Trees.scala:1436)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
	at scala.reflect.internal.Trees.itransform(Trees.scala:1411)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
	at scala.reflect.internal.Trees.itransform(Trees.scala:1430)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
	at scala.reflect.internal.Trees.itransform(Trees.scala:1409)
	at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
	at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
	at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
	at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
```

**Comment by rfoley-fw** [comment]
Thanks for the details.  We will take a look and get back to you.

**Comment by rfoley-fw** [comment]
From the errors, it looks like the Spark build is failing.  

I followed the steps from the README.md to build the repo, and it is succeeding.  Below we can see the log from it completing successfully.
Below are the steps used to build.

git clone https://github.com/open-infrastructure-labs/caerus-dike.git
cd caerus-dike/
git checkout tags/v1.0.0
git submodule init
git submodule update --recursive --progress
 ./build.sh

If you are still seeing the issue, can you please list the steps being used to build?  Thank you !

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18:39 min
[INFO] Finished at: 2021-11-08T15:40:45Z
[INFO] ------------------------------------------------------------------------

**Comment by pramodhrachuri** [comment]
Hi,
I have used exactly the same commands as you suggested and the build is still failing with the same error. The following is the tail of my shell history.
```
  685  git clone https://github.com/open-infrastructure-labs/caerus-dike.git
  686  cd caerus-dike/
  687  git checkout tags/v1.0.0
  688  git submodule init
  689  git submodule update --recursive --progress
  690  ./build.sh
  691  history | tail
```
I have added the important part of the build log below.

Edit: I got the error on 4/6 servers and the build is still running on the remaining 2 servers.

Thanks
-Pramodh

```
[INFO] ------------------------------------------------------------------------                                                                                              
[INFO] Reactor Summary for Spark Project Parent POM 3.3.0-SNAPSHOT:                                                                                                          
[INFO]                                                                                                                                                                       
[INFO] Spark Project Parent POM ........................... SUCCESS [01:05 min]                                                                                              
[INFO] Spark Project Tags ................................. SUCCESS [ 25.190 s]                                                                                              
[INFO] Spark Project Sketch ............................... SUCCESS [ 10.124 s]                                                                                              
[INFO] Spark Project Local DB ............................. SUCCESS [  8.343 s]                                                                                              
[INFO] Spark Project Networking ........................... SUCCESS [ 12.503 s]                                                                                              
[INFO] Spark Project Shuffle Streaming Service ............ SUCCESS [  8.295 s]                                                                                              
[INFO] Spark Project Unsafe ............................... SUCCESS [ 16.350 s]                                                                                              
[INFO] Spark Project Launcher ............................. SUCCESS [  9.032 s]                                                                                              
[INFO] Spark Project Core ................................. SUCCESS [04:00 min]                                                                                              
[INFO] Spark Project ML Local Library ..................... SUCCESS [ 39.056 s]                                                                                              
[INFO] Spark Project GraphX ............................... SUCCESS [ 44.657 s]                                                                                              
[INFO] Spark Project Streaming ............................ SUCCESS [01:20 min]                                                                                              
[INFO] Spark Project Catalyst ............................. FAILURE [01:07 min]                                                                                              
[INFO] Spark Project SQL .................................. SKIPPED                                                                                                          
[INFO] Spark Project ML Library ........................... SKIPPED                                                                                                          
[INFO] Spark Project Tools ................................ SKIPPED                                                                                                          
[INFO] Spark Project Hive ................................. SKIPPED                                                                                                          
[INFO] Spark Project REPL ................................. SKIPPED                                                                                                          
[INFO] Spark Project Assembly ............................. SKIPPED                                                                                                          
[INFO] Kafka 0.10+ Token Provider for Streaming ........... SKIPPED                                                                                                          
[INFO] Spark Integration for Kafka 0.10 ................... SKIPPED                                                                                                          
[INFO] Kafka 0.10+ Source for Structured Streaming ........ SKIPPED                                                                                                          
[INFO] Spark Project Examples ............................. SKIPPED                                                                                                          
[INFO] Spark Integration for Kafka 0.10 Assembly .......... SKIPPED                                                                                                          
[INFO] Spark Avro ......................................... SKIPPED                                                                                                          
[INFO] ------------------------------------------------------------------------                                                                                              
[INFO] BUILD FAILURE                                                                                                                                                         
[INFO] ------------------------------------------------------------------------                                                                                              
[INFO] Total time:  10:29 min                                                                                                                                                
[INFO] Finished at: 2021-11-08T17:27:59Z                                                                                                                                     
[INFO] ------------------------------------------------------------------------
---------------------------------------------------
constituent[0]: file:/usr/share/maven/conf/logging/
constituent[1]: file:/usr/share/maven/lib/sisu-plexus.jar
constituent[2]: file:/usr/share/maven/lib/maven-plugin-api-3.x.jar
constituent[3]: file:/usr/share/maven/lib/jcl-over-slf4j.jar
constituent[4]: file:/usr/share/maven/lib/maven-resolver-connector-basic.jar
constituent[5]: file:/usr/share/maven/lib/maven-resolver-transport-wagon.jar
constituent[6]: file:/usr/share/maven/lib/maven-compat-3.x.jar
constituent[7]: file:/usr/share/maven/lib/maven-slf4j-provider-3.x.jar
constituent[8]: file:/usr/share/maven/lib/maven-resolver-api.jar
constituent[9]: file:/usr/share/maven/lib/guice.jar
constituent[10]: file:/usr/share/maven/lib/plexus-component-annotations.jar
constituent[11]: file:/usr/share/maven/lib/commons-lang3.jar
constituent[12]: file:/usr/share/maven/lib/plexus-utils.jar
constituent[13]: file:/usr/share/maven/lib/maven-core-3.x.jar
constituent[14]: file:/usr/share/maven/lib/maven-shared-utils.jar
constituent[15]: file:/usr/share/maven/lib/jsr250-api.jar
constituent[16]: file:/usr/share/maven/lib/maven-model-builder-3.x.jar
constituent[17]: file:/usr/share/maven/lib/sisu-inject.jar
constituent[18]: file:/usr/share/maven/lib/plexus-cipher.jar
constituent[19]: file:/usr/share/maven/lib/maven-embedder-3.x.jar
constituent[20]: file:/usr/share/maven/lib/cdi-api.jar
constituent[21]: file:/usr/share/maven/lib/plexus-interpolation.jar
constituent[22]: file:/usr/share/maven/lib/wagon-file.jar
constituent[23]: file:/usr/share/maven/lib/slf4j-api.jar
constituent[24]: file:/usr/share/maven/lib/maven-settings-builder-3.x.jar
constituent[25]: file:/usr/share/maven/lib/maven-model-3.x.jar
constituent[26]: file:/usr/share/maven/lib/maven-resolver-util.jar
constituent[27]: file:/usr/share/maven/lib/commons-cli.jar
constituent[28]: file:/usr/share/maven/lib/maven-builder-support-3.x.jar
constituent[29]: file:/usr/share/maven/lib/maven-resolver-impl.jar
constituent[30]: file:/usr/share/maven/lib/wagon-provider-api.jar
constituent[31]: file:/usr/share/maven/lib/jansi.jar
constituent[32]: file:/usr/share/maven/lib/commons-io.jar
constituent[33]: file:/usr/share/maven/lib/maven-settings-3.x.jar
constituent[34]: file:/usr/share/maven/lib/maven-resolver-provider-3.x.jar
constituent[35]: file:/usr/share/maven/lib/javax.inject.jar
constituent[36]: file:/usr/share/maven/lib/guava.jar
constituent[37]: file:/usr/share/maven/lib/wagon-http-shaded.jar
constituent[38]: file:/usr/share/maven/lib/maven-artifact-3.x.jar
constituent[39]: file:/usr/share/maven/lib/maven-resolver-spi.jar
constituent[40]: file:/usr/share/maven/lib/maven-repository-metadata-3.x.jar
constituent[41]: file:/usr/share/maven/lib/aopalliance.jar
constituent[42]: file:/usr/share/maven/lib/plexus-sec-dispatcher.jar
---------------------------------------------------
Exception in thread "main" java.lang.StackOverflowError
        at scala.reflect.internal.Trees.itransform(Trees.scala:1402)
        at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
        at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
        at scala.reflect.internal.Trees.itransform(Trees.scala:1409)
        at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
        at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
        at scala.reflect.internal.Trees.itransform(Trees.scala:1436)
        at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
        at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
        at scala.reflect.internal.Trees.itransform(Trees.scala:1411)
        at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.api.Trees$Transformer.transform(Trees.scala:2563)
        at scala.tools.nsc.transform.TypingTransformers$TypingTransformer.transform(TypingTransformers.scala:57)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:275)
        at scala.tools.nsc.transform.ExtensionMethods$Extender.transform(ExtensionMethods.scala:133)
        at scala.reflect.internal.Trees.itransform(Trees.scala:1430)
        at scala.reflect.internal.Trees.itransform$(Trees.scala:1400)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
        at scala.reflect.internal.SymbolTable.itransform(SymbolTable.scala:28)
```

**Comment by pramodhrachuri** [comment]
The build got completed on two servers but not on the remaining 4 servers. I was able to build on all six servers before. Any suggestions on debugging and what to compare between the servers?

I checked git's and docker's version numbers and even ~purged~ pruned all the container images on the servers before building. For every trial, I am cloning the repo again instead of reusing previous pulls. Still no luck.

Thanks
-Pramodh

**Comment by rfoley-fw** [comment]
Thanks for the details !  This build failure during the Spark build is not something that we have seen before.  We are using the Spark build as is, without modification, so we are subject to all the same issues that a Spark build can see.  We tried reproducing this today on a few different machines, but no luck so far.  

We have a few ideas for workarounds to the Spark build issue, and are actively looking a this.  We will get back to you with more details.

**Comment by rfoley-fw** [comment]
We are still in the process of looking at a workaround approach.  Could you please provide a few more details on both the machines that successfully compiled and those that failed?  We are just trying to understand how these machine configs compared to the machines we are using.
The information we are interested in is:
- Machine details (CPU type, memory size)
- Operating System and Version
- Kernel Version
- Docker Version
Thanks Very Much !

**Comment by pramodhrachuri** [comment]
Thanks for your support! We found that sometimes removing the folder and running the commands you sent again works. After several reattempts, the build is complete on all the machines now. I am not sure why that is happening but it worked. Maybe there is a race condition during the compilation.

I have listed down the specifications of the machines and the number of attempts it took below.

| SNo |     Processor    | Hyperthreading | Memory  (RAM + Swap) |       OS       |   Kernel   | Docker Version | Number of Attempts |
|:----------:|:----------------:|:--------------:|:--------------------:|:--------------:|:----------:|:--------------:|:------------------:|
|          1 | Intel Xeon L5520 | ON             | 31GB +  1 GB         | Ubuntu 18.04   | 4.15.0-20-generic  | 19.03.12       | 4                  |
|          2 | Intel Xeon L5520 | ON             | 31GB +  1 GB         | Ubuntu 18.04.6 | 4.15.0-158-generic | 20.10.8        | 2                  |
|          3 | Intel Xeon L5520 | ON             | 27GB +  1 GB         | Ubuntu 18.04   | 4.15.0-20-generic  | 19.03.12       | 3                  |
|          4 | Intel Xeon L5520 | ON             | 31GB + 31 GB         | Ubuntu 18.04.6 | 4.15.0-159-generic | 20.10.7        | 2                  |
|          5 | Intel Xeon L5520 | ON             | 23GB + 31 GB         | Ubuntu 18.04.6 | 4.15.0-159-generic | 20.10.7        | 1                  |
|          6 | Intel Xeon L5520 | ON             | 11GB +  0 GB         | Ubuntu 18.04.6 | 4.15.0-159-generic | 20.10.7        | 1                  |

**Comment by rfoley-fw** [comment]
Thanks for the details !  This is very helpful !

We did finish the workaround and have published a new release v1.1.0.
This workaround essentially eliminates the Spark build from the build process.  Now that the code we rely on has made it into a release, we can eliminate the Spark build from our setup.  We now simply download the released Spark binaries and install them for our use.
In addition to being simpler, this also should make the build quite a bit quicker.

**Comment by pramodhrachuri** [comment]
That's great! I will start using v1.1.0 from now on.

Thank you very much!
